### PR TITLE
[serdes] move most of the models over to using spec

### PR DESCRIPTION
**Pull request by sansolovyov**
Refs #43068

It moves most of the models to using spec, others need a bit more involvement so I want them to be separate so it's not too messy. It's easier to read every commit separately, they are quite self-contained.

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/models/entity_id_test.clj [hunk 1]
```diff
@@ -60,6 +60,7 @@
     :model/PermissionsGroupMembership
     :model/PermissionsRevision
     :model/PersistedInfo
+    :model/Pulse
```
Comment (crisptrutski) [inline enterprise/backend/test/metabase_enterprise/models/entity_id_test.clj:63]: This was tragic. It feels like a test should have caught this, perhaps something like testing that this is the union of `v2.models/excluded-models` and `inlined-models` (and that those are disjoint)?

**Review by crisptrutski — APPROVED** [review]
This is so beautiful 🥹

File: src/metabase/models/timeline_event.clj [hunk 2]
```diff
@@ -116,11 +115,12 @@
...
-      (update :created_at  #(if (string? %) (u.date/parse %) %))))
+
+;; nested in Timeline
+(defmethod serdes/make-spec "TimelineEvent" [_model-name _opts]
+  {:copy      [:archived :description :icon :name :time_matters :timezone]
+   :skip      []
+   :transform {:created_at  (serdes/date)
+               :creator_id  (serdes/fk :model/User)
+               :timeline_id (serdes/parent-ref)
+               :timestamp   (serdes/date)}})
```
[the remark this answers is not in the document]
Comment (piranha) [reply src/metabase/models/timeline_event.clj:126]: Yeah I thought it looks nice; could be a var, but then you need to know what is a var and what is a function...

File: enterprise/backend/test/metabase_enterprise/models/entity_id_test.clj (hunk 1 above)
Comment (piranha) [reply enterprise/backend/test/metabase_enterprise/models/entity_id_test.clj:63]: uhhh this is hard and a bit disjointed :\

**Review by crisptrutski — APPROVED** [review]
Re-approving after most recent commit

File: enterprise/backend/src/metabase_enterprise/serialization/v2/models.clj [hunk 3]
```diff
@@ -32,7 +31,8 @@
    "Dimension"
    "ParameterCard"
    "DashboardCardSeries"
-   "TimelineEvent"])
+   "TimelineEvent"
+   "FieldValues"])
```
Comment (crisptrutski) [inline enterprise/backend/src/metabase_enterprise/serialization/v2/models.clj:35]: But FieldValues is not inlined?

Comment (piranha) [reply enterprise/backend/src/metabase_enterprise/serialization/v2/models.clj]: It's not, was a mistake 😳

**Pull request merged**